transport/transport.go

```diff
@@ -225,6 +241,8 @@ func (t *Transport) run(ctx context.Context) {
 			enc = codec.LengthPrefixEncoder(codec.PlainEncoder, enc)
 			dec = codec.LengthPrefixDecoder(codec.PlainDecoder, dec)
 
+			t.table.AddIP(remote, addr[:strings.IndexByte(addr, ':')])
```

Comment:
You need to add a test (or check to an existing test) that demonstrates the transport table has the IP of the client after the client has connected / sent a message. This ensures future modifications do not break the assumption that this is being done.